cml-core/src/core/inference.rs

```diff
@@ -15,15 +15,15 @@ pub struct NewSample<F> {
     optional_tags: Option<Vec<F>>,
 }
 
-pub trait Inference<M, F, C: Manager> {
+pub trait Inference<M, F, T, C: Manager> {
     async fn init_inference(
         &self,
         target_type: M,
         optional_fields: Option<Vec<M>>,
         optional_tags: Option<Vec<M>>,
     ) -> Result<()>;
 
-    async fn inference<FN, R>(
+    async fn inference<FN>(
```

Comment:
I think here we need a parameter available_status, just like working_status in task.rs, which allows user to custom the list of available status. For example, one could mark some status for a specific type of inference, but use others for another inference operation.